lib/posix/posix_other.nim

```diff
@@ -140,12 +140,15 @@ type
     ## used for block sizes
   Clock* {.importc: "clock_t", header: "<sys/types.h>".} = int
   ClockId* {.importc: "clockid_t", header: "<sys/types.h>".} = int
-  Dev* {.importc: "dev_t", header: "<sys/types.h>".} = int
+  Dev* {.importc: "dev_t", header: "<sys/types.h>".} = uint64
```

Comment:
Please fork this file into `posix_linux_i386`, or you can adapt `posix_linux_amd64` to work for 32bit. I'd say `posix_other` should just be kept as a template for future OSes.

Also following Linux's definition is not wise, especially since this file is used by many other OS that doesn't have their own dedicated version.

Ideally we need a tool to auto-detect these types (complementary to `tools/detect`) so that contributors can add support for other POSIX OS faster.